solution/text-extractor/text_extractor.py

```diff
@@ -42,55 +42,53 @@ def handler(event: dict, context: dict) -> dict:
         token = config["token"]
     except KeyError:
         return lambda_response(400, "You must include 'id', 'uri', and 'post_url' in the request body.")
+    with TemporaryDirectory() as temp_dir:
+        # Retrieve the file
+        try:
+            backend_type = config.get("backend", "web").lower()
+            backend = FileBackend.get_backend(backend_type, config)
+            file = backend.get_file(temp_dir, uri)
+        except BackendInitException as e:
+            return lambda_response(500, f"Failed to initialize backend: {str(e)}")
+        except BackendException as e:
+            return lambda_response(500, f"Failed to retrieve file: {str(e)}")
+        except Exception as e:
+            return lambda_response(500, f"Backend unexpectedly failed: {str(e)}")
 
-    # Retrieve the file
-    try:
-        backend_type = config.get("backend", "web").lower()
-        backend = FileBackend.get_backend(backend_type, config)
-        file = backend.get_file(uri)
-    except BackendInitException as e:
-        return lambda_response(500, f"Failed to initialize backend: {str(e)}")
-    except BackendException as e:
-        return lambda_response(500, f"Failed to retrieve file: {str(e)}")
-    except Exception as e:
-        return lambda_response(500, f"Backend unexpectedly failed: {str(e)}")
-
-    # Determine the file's MIME type
-    # Magic docs recommend using first 2048 bytes of the file for most accurate type detection
-    try:
-        file_type = magic.from_buffer(file[:2048], mime=True)
-    except Exception as e:
-        return lambda_response(500, f"Failed to determine file type: {str(e)}")
+        try:
+            file_type = uri.split('.')[-1]
```

Comment:
Mime types have a hard time with smaller documents.  We are just going to assume the extension is correct instead.